### ISSUE TITLE
Task completed callbacks

### ISSUE THREAD

**Issue by ismellike** [Lay3rLabs/avs-toolkit#76]
Once a task is completed, we should be able to perform actions based on the given result.

![image](https://github.com/user-attachments/assets/e424426a-887c-4170-aedc-0960a3664cf1)

**Comment by JakeHartnell** [comment]
This would be great for the hackathon. A few good use cases that could use it.

**Comment by JakeHartnell** [comment]
Also, I think it would be sick if a TaskCompleted hook could create a new Task! : )

Stretch goal, doesn't need to be production ready as it's proof of concept for a hackathon.

**Comment by JakeHartnell** [comment]
Had a conversation with @ethanfrey and we could potentially do these a different way. Where we can define a callback when we create the task... not sure all the implications of this, but that could be very useful for oracles in particular (i.e. get the latest data and then do X).

Worth discussing more but even something basic like we described above and in the draft PR is good enough for demos.

**Comment by ethanfrey** [comment]
Rather than "draft PR for demos" (cuz we have a few weeks til next event), it would be good to get proper requirements and design it.

Including one or two concrete demos that should be supported and the general requirements would be a good start (rather than a technical implementation of which hooks to add).

**Comment by ethanfrey** [comment]
I would love if all these "hacks for some demos" could be in different repos.
Otherwise all the hacky code ends up in production.

They could be as "spikes" maybe but then clearly marked as never to be merged.
I think new features should go through a proper design process, so we have a clean API and maintainability as this grows up.

**Comment by NoahSaso** [comment]
> Had a conversation with @ethanfrey and we could potentially do these a different way. Where we can define a callback when we create the task... not sure all the implications of this, but that could be very useful for oracles in particular (i.e. get the latest data and then do X).
> 
> Worth discussing more but even something basic like we described above and in the draft PR is good enough for demos.

yeah, defining callback on creation is probably the way to go IMO. that's how [polytone works](https://github.com/cosmos/ibc-apps/blob/main/cosmwasm/polytone/contracts/main/note/src/msg.rs#L48-L53), and it gives the caller full control over a task's behavior.

**Comment by ethanfrey** [comment]
I'm going to merge this as we need something stable for the devcon demos.
Longer-term, we need to do a proper design session here and probably do a lot of breaking changes.
